### PR TITLE
fix(osqp_interface): modify build error in rolling

### PR DESCRIPTION
**Pull request by wep21**
## Description

This PR fixes the following build error on rolling distro.
```
gmake[1]: *** [CMakeFiles/Makefile2:157: CMakeFiles/osqp_interface.dir/all] エラー 2
gmake: *** [Makefile:146: all] エラー 2
--- stderr: osqp_interface                             
In file included from /home/daisuke/workspace/autoware.proj.universe.rolling/src/autoware/universe/common/osqp_interface/src/osqp_interface.cpp:15:
/home/daisuke/workspace/autoware.proj.universe.rolling/src/autoware/universe/common/osqp_interface/include/osqp_interface/osqp_interface.hpp:20:10: fatal error: osqp/osqp.h: そのようなファイルやディレクトリはありません
   20 | #include "osqp/osqp.h"
      |          ^~~~~~~~~~~~~
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/osqp_interface/CMakeLists.txt [hunk 1]
```diff
@@ -19,14 +19,16 @@ project(osqp_interface)
...
-get_target_property(OSQP_INCLUDE_DIR osqp::osqp INTERFACE_INCLUDE_DIRECTORIES)
 
 ament_auto_find_build_dependencies(REQUIRED
   ${${PROJECT_NAME}_BUILD_DEPENDS}
   ${${PROJECT_NAME}_BUILDTOOL_DEPENDS}
 )
 
+find_package(osqp REQUIRED)
+get_target_property(OSQP_INCLUDE_SUB_DIR osqp::osqp INTERFACE_INCLUDE_DIRECTORIES)
+get_filename_component(OSQP_INCLUDE_DIR ${OSQP_INCLUDE_SUB_DIR} PATH)
```
Comment (kenji-miyake) [inline common/osqp_interface/CMakeLists.txt:30]: Could you explain this a bit more in detail? :pray: 
If it should be after `ament_auto_find_build_dependencies`, it's good to add some comments.

Comment (wep21) [reply common/osqp_interface/CMakeLists.txt:30]: I guess we should find `osqp` after finding `osqp_vendor` in `ament_auto_find_build_dependencies` .

Comment (kenji-miyake) [reply common/osqp_interface/CMakeLists.txt:30]: I see, then could you add the comment? :pray:

Comment (wep21) [reply common/osqp_interface/CMakeLists.txt:31]: I added comment at https://github.com/autowarefoundation/autoware.universe/pull/776/commits/063fc4247ffce699198c98e0f995b055f4d6a340.

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**